nrpy/infrastructures/BHaH/seobnr/SEOBNR_C_codegen_library.py

```diff
@@ -184,31 +204,100 @@ def register_CFunction_SEOBNRv5_aligned_spin_ode_integration() -> (
 = gsl_odeiv2_step_alloc (T, 4);
 gsl_odeiv2_control *restrict c
 = gsl_odeiv2_control_standard_new(1e-12, 1e-11, 1.0, 1.0);
-gsl_odeiv2_evolve *restrict e
-= gsl_odeiv2_evolve_alloc (4);
-
 gsl_odeiv2_system sys = {SEOBNRv5_aligned_spin_right_hand_sides, NULL, 4, commondata};
 
-double t = 0.0;
-const double t1 = 1e6;
-double h = 1e-6;
-double y[4];
+REAL t = 0.0;
+REAL t_new;
+const REAL t1 = 2e9;
+const int dyn_size = 8; //t,r,phi,prstar,pphi,Hreal, Omega, Omega_circ
+REAL y[4],yerr[4],dydt_in[4],dydt_out[4];
+int status = 0;
+int i;
+int stop = 0;
 y[0] = commondata->r;
 y[1] = commondata->phi;
 y[2] = commondata->prstar;
 y[3] = commondata->pphi;
+status = SEOBNRv5_aligned_spin_right_hand_sides(t,y,dydt_in,commondata);
+SEOBNRv5_aligned_spin_augments(commondata);
+REAL h = 2.0*M_PI/dydt_in[1]/5.0;
+size_t bufferlength = (size_t)(t1/h); // runs up to 0.01x maximum time (we should not ideally run that long) 
+commondata->dynamics = (REAL *)calloc(bufferlength*(dyn_size),sizeof(REAL));
+commondata->nsteps_low = 0;
+
+//store
+commondata->dynamics[dyn_size*commondata->nsteps_low] = t;
+for (i = 0; i < 4; i++){
+  commondata->dynamics[dyn_size*commondata->nsteps_low + 1 + i] = y[i];
+}
+commondata->dynamics[dyn_size*commondata->nsteps_low + 5] = commondata->Hreal;
+commondata->dynamics[dyn_size*commondata->nsteps_low + 6] = dydt_in[1];
+commondata->dynamics[dyn_size*commondata->nsteps_low + 7] = commondata->Omega_circ;
+commondata->nsteps_low++;
+
+
+while(stop == 0) {
+  //integrate
+  status = gsl_odeiv2_step_apply (s, t, h, y, yerr, dydt_in, dydt_out, &sys);
+  if (status != GSL_SUCCESS){
+    printf("Error in step_apply!\\n");
+    return status;
+  }
+  t_new = t + h;
```

Comment:
This seems to be mispositioned; h for the coming timestep is set in the next line.

Reply:
Hi Zach, 

Thanks for catching it.
I have updated the code.